Classes/ExceptionHandler/ProductionExceptionHandler.php

```diff
@@ -0,0 +1,63 @@
+<?php
+namespace PunktDe\PtExtbase\ExceptionHandler;
+
+/*
+ * This file is part of the PunktDe\PtExtbase package.
+ *
+ * This package is open source software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use TYPO3\CMS\Core\Error\ProductionExceptionHandler as Typo3ProductionExceptionHandler;
+use TYPO3\CMS\Core\Messaging\AbstractStandaloneMessage;
+use TYPO3\CMS\Core\Messaging\ErrorpageMessage;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+class ProductionExceptionHandler extends Typo3ProductionExceptionHandler
+{
+    const CONTEXT_AJAX = 'AJAX';
+
+    /**
+     * @param \Exception|\Throwable $exception
+     */
+    public function echoExceptionWeb($exception)
+    {
+        if (isset($GLOBALS['TYPO3_AJAX']) && $GLOBALS['TYPO3_AJAX'] === true) {
+            echo $exception->getCode();
```

Comment:
In dem Fall sollten doch auch Status Headers gesendet werden?

Reply:
gute Frage - wir setzen die Status Header vorher, und wollen nicht mehr dass die überschrieben werden. TYPO3 setzt in der Funktion `sendStatusHeaders` blind einen 500, es sei denn die Exception ist eine `TYPO3\CMS\Core\Error\Http\StatusException`. Soweit ich sehe gibt es keine Möglichkeit zu prüfen, ob ein Header schon gesetzt ist, nur eine, ob die Header schon gesendet sind - und dann wär eh zu spät. Lass uns da mal nochmal drüber schauen ob man das intelligent lösen kann